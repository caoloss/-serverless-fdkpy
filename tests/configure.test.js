const fdk = require('../index')
const eventGatewayProcesses = require('./event-gateway/processes')

const config = {
  functions: [
    {
      functionId: 'myFunctionOne',
      provider: {
        type: 'awslambda',
        arn: 'https://xxxxxxxxxx.execute-api.us-east-1.amazonaws.com/dev/test',
        region: 'us-east-1',
      },
    },
    {
      functionId: 'myFunctionTwo',
      provider: {
        type: 'http',
        url: 'http://www.example.com',
      },
    },
  ],
  subscriptions: [
    { functionId: 'myFunctionOne', event: 'pageVisited' },
    { functionId: 'myFunctionOne', event: 'userCreated' },
  ],
}

let eventGateway
let eventGatewayProcessId

beforeAll(() =>
  eventGatewayProcesses
    .spawn({
      configPort: 4017,
      apiPort: 4018,
    })
    .then(processInfo => {
      eventGatewayProcessId = processInfo.id
      eventGateway = fdk.createEventGatewayClient({
        hostname: 'localhost',
        configurationProtocol: 'http',
        configurationPort: 4017,
      })
    })
)

afterAll(() => {
  eventGatewayProcesses.shutDown(eventGatewayProcessId)
})

test('should created multiple functions and subscriptions with configure', () => {
  expect.assertions(1)
  return eventGateway.configure(config).then(response => {
    expect(response).toMatchSnapshot()
  })
})
