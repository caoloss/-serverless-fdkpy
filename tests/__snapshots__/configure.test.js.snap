// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should created multiple functions and subscriptions with configure 1`] = `
Object {
  "functions": Array [
    Object {
      "functionId": "myFunctionOne",
      "provider": Object {
        "arn": "https://xxxxxxxxxx.execute-api.us-east-1.amazonaws.com/dev/test",
        "region": "us-east-1",
        "type": "awslambda",
      },
    },
    Object {
      "functionId": "myFunctionTwo",
      "provider": Object {
        "type": "http",
        "url": "http://www.example.com",
      },
    },
  ],
  "subscriptions": Array [
    Object {
      "event": "pageVisited",
      "functionId": "myFunctionOne",
      "subscriptionId": "pageVisited-myFunctionOne",
    },
    Object {
      "event": "userCreated",
      "functionId": "myFunctionOne",
      "subscriptionId": "userCreated-myFunctionOne",
    },
  ],
}
`;
